frontend/src/pages/projects/screens/detail/pipelines/PipelinesList.tsx

```diff
@@ -9,11 +9,19 @@ import { usePipelinesAPI } from '~/concepts/pipelines/context';
 import EmptyStateErrorMessage from '~/components/EmptyStateErrorMessage';
 import { TABLE_CONTENT_LIMIT, LIMIT_MAX_ITEM_COUNT } from '~/concepts/pipelines/const';
 
-const PipelinesList: React.FC = () => {
+interface PipelinesListProps {
+  setIsPipelinesEmpty: (arg0: boolean) => void;
+}
+
+const PipelinesList: React.FC<PipelinesListProps> = ({ setIsPipelinesEmpty }) => {
   const { namespace } = usePipelinesAPI();
   const [pipelines, loaded, loadError, refresh] = usePipelines(LIMIT_MAX_ITEM_COUNT);
   const navigate = useNavigate();
 
+  React.useEffect(() => {
+    pipelines.length === 0 ? setIsPipelinesEmpty(true) : setIsPipelinesEmpty(false);
+  }, [pipelines, setIsPipelinesEmpty]);
```

Comment:
When it comes to effects that execute a callback, you want to limit the callback to only notify when the value has changes. In this case, only when going from `true => false` and from `false => true`.

```suggestion
  const isPipelinesEmpty = pipelines.length === 0;
  React.useEffect(() => {
     setIsPipelinesEmpty(isPipelinesEmpty);
  }, [isPipelinesEmpty, setIsPipelinesEmpty]);
```

For such callbacks, I don't think that `setIsPipelinesEmpty` should be part of the hook dependency but it doesn't hurt here and so we may as well keep it, or get into a larger discussion about hook dependencies :)